gdal/frmts/netcdf/netcdfsgwriterutil.cpp

```diff
@@ -0,0 +1,910 @@
+        {
+            OGRLineString* as_line_ref = dynamic_cast<OGRLineString*>(this->geometry_ref);
+            as_line_ref->getPoint(point_index, &pt_buffer);
+        }
+
+        if (this->type == MULTILINE)
+        {
+            OGRMultiLineString* as_mline_ref = dynamic_cast<OGRMultiLineString*>(this->geometry_ref);
+            int part_ind = static_cast<int>(part_no);
+            OGRLineString* lstring = dynamic_cast<OGRLineString*>(as_mline_ref->getGeometryRef(part_ind));
+            lstring->getPoint(point_index, &pt_buffer);
+        }
+
+        if (this->type == POLYGON)
+        {
+            OGRPolygon* as_polygon_ref = dynamic_cast<OGRPolygon*>(this->geometry_ref);
+                        int ring_ind = static_cast<int>(part_no);
+
+            if(part_no == 0)
+            {
+                as_polygon_ref->getExteriorRing()->getPoint(point_index, &pt_buffer);
+            }
+
+            else
+            {
+                as_polygon_ref->getInteriorRing(ring_ind - 1)->getPoint(point_index, &pt_buffer);
+            }
+        }
+
+        if (this->type == MULTIPOLYGON)
+        {
+            OGRMultiPolygon* as_mpolygon_ref = dynamic_cast<OGRMultiPolygon*>(this->geometry_ref);
+
+            int polygon_num = 0;
+            int ring_number = 0;
+            int pno_itr = static_cast<int>(part_no);
+
+            // Find the right polygon, and the right ring number
+            for(int pind = 0; pind < as_mpolygon_ref->getNumGeometries(); pind++)
+            {
+                OGRPolygon * itr_poly = dynamic_cast<OGRPolygon*>(as_mpolygon_ref->getGeometryRef(pind));
+                if(pno_itr < (itr_poly->getNumInteriorRings() + 1)) // + 1 is counting the EXTERIOR ring
+                {
+                    ring_number = static_cast<int>(pno_itr);
+                    break;
+                }
+
+                else
+                {
+                    pno_itr -= (itr_poly->getNumInteriorRings() + 1);
+                    polygon_num++;
+                }
+            }
+
+            OGRPolygon* key_polygon = dynamic_cast<OGRPolygon*>(as_mpolygon_ref->getGeometryRef(polygon_num));
+
+            if(ring_number == 0)
+            {
+                key_polygon->getExteriorRing()->getPoint(point_index, &pt_buffer);
+            }
+
+            else
+            {
+                key_polygon->getInteriorRing(ring_number - 1)->getPoint(point_index, &pt_buffer);
+            }
+        }
+
+        return pt_buffer;
+    }
+
+    void OGR_SGeometry_Scribe::writeSGeometryFeature(SGeometry_Feature& ft)
+    {
+        if (ft.getType() == NONE)
+        {
+            throw SG_Exception_BadFeature();
+        }
+
+        // Write each point from each part in node coordinates
+        for(size_t part_no = 0; part_no < ft.getTotalPartCount(); part_no++)
+        {
+            if(ft.getType() == POLYGON || ft.getType() == MULTIPOLYGON)
+            {
+                int interior_ring_fl = 0;
```

Comment:
Oh the purpose of using an int here is actually because interior_ring_fl is used in nc_put_var1_int (or whatever it's called) later. Explicitly the standard needs this to be = 1 or = 0 written as an int.

Reply:
As far as I can see, this variable is not used other than setting it to 0 or 1, or testing it. Not passed to nc_put_var1_int ()

Reply:
Oh yeah you're right, this is left over from when I _did_ write it here but that was from some older version of this. Thanks, nice catch :+1: I'll change it to bool